src/portalnet/discovery.rs

```diff
@@ -35,7 +37,12 @@ pub struct Discovery {
 
 impl Discovery {
     pub fn new(config: Config) -> Result<Self, String> {
-        let enr_key = CombinedKey::generate_secp256k1();
+        let enr_key = match config.private_key {
+            Some(val) => {
+                CombinedKey::secp256k1_from_bytes(val.clone().as_mut_slice()).unwrap()
```

Comment:
This pr took way longer than it should have, but most of that time was wasted b/c I was using `b'0' * 32` as my test private key, but for whatever reason this library rejects the 0 bytestring as an invalid private key. Is this just a security thing? Is this desired behavior? `b'0' * 31 + b'1'` is accepted, along with all other private keys i've used

Reply:
Yup, I don't remember the details of why, but 0 is an invalid key. Rejecting that key is the correct behavior AIUI.